src/Sarif/Readers/RuleDictionaryConverter.cs

```diff
@@ -6,9 +6,13 @@
 
 namespace Microsoft.CodeAnalysis.Sarif.Readers
 {
-    public class IRuleDictionaryConverter : JsonConverter
+    /// <summary>
+    /// This converter only exists because the version of Json.NET we're using (6.0.8)
+    /// can't write out a property whose static type is an interface.
+    /// </summary>
+    public class RuleDictionaryConverter : JsonConverter
     {
-        public static readonly IRuleDictionaryConverter Instance = new IRuleDictionaryConverter();
+        public static readonly RuleDictionaryConverter Instance = new RuleDictionaryConverter();
```

Comment:
I don't see that in "Framework Design Guidelines," 2nd Ed., Sec. 3.5.2, "Names of common types". It just says

IDictionary or IDictionary<TKey, TValue>: DO add the suffix "Dictionary".

---
In reply to: [63250824](https://github.com/Microsoft/sarif-sdk/pull/270#discussion_r63250824) [](ancestors = 63250824,63249962)